src/test/java/io/spokestack/spokestack/wakeword/RingBufferTest.java

```diff
@@ -98,6 +98,14 @@ public void testSeek() {
         for (int i = 0; i < buffer.capacity(); i++)
             buffer.write(i + 1);
 
+        // negative seek wraps around --
+        // position 5 in the buffer contains 0;
+        // we'll end up at position 5, which contains 5
+        buffer.seek(-2);
```

Comment:
this seek is undefined, because the read head is at the beginning of the valid region of the buffer
a better test would be to read/seek to a valid offset and then execute a negative seek